### PR TITLE
Bug 1321414 -Create v2 desktop churn/retention dataset in telemetry-parquet

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This continues the discussion from [this pull request](https://github.com/mozilla-services/data-pipeline/pull/236).

In short, this prepares the churn dataset for being scheduled on airflow and to be available on redash. The last thing to do is to change the s3 paths.

I've compared the results in the base version of this notebook (with the submission_date_s3 patch) with the results from this notebook. There are no significant changes in results from the original script. All changed rows are the result of small rounding errors. 

```
Diff of original and patched in daterange 20161113-20161119
+ csvdiff --style=summary channel,geo,is_funnelcake,acquisition_period,start_version,start_version,sync_usage,current_version,current_week,is_active churn-20161113-20161119.by_activity_original.csv churn-20161113-20161119.by_activity.csv
1 rows removed (0.0%)
1 rows added (0.0%)
930 rows changed (0.0%)

Diff of original and patched in daterange 20161120-20161126
+ csvdiff --style=summary channel,geo,is_funnelcake,acquisition_period,start_version,start_version,sync_usage,current_version,current_week,is_active churn-20161120-20161126.by_activity_original.csv churn-20161120-20161126.by_activity.csv
1 rows removed (0.0%)
1 rows added (0.0%)
949 rows changed (0.0%)

Diff of original and patched in daterange 20161127-20161203
+ csvdiff --style=summary channel,geo,is_funnelcake,acquisition_period,start_version,start_version,sync_usage,current_version,current_week,is_active churn-20161127-20161203.by_activity_original.csv churn-20161127-20161203.by_activity.csv
1 rows removed (0.0%)
1 rows added (0.0%)
1133 rows changed (0.0%)
```

On a side note, itts interesting to note that the change to using submission_date_s3 causes roughly a 1% difference in results.

```
csvdiff --style=summary channel,geo,is_funnelcake,acquisition_period,start_version,start_version,sync_usage,current_version,current_week,is_active churn-20161009-20161015.by_activity_mreid.csv
churn-20161009-20161015.by_activity_testing.csv

21395 rows removed (1.0%)
25274 rows added (1.2%)
26267 rows changed (1.3%)
```

r? @Dexterp37 @mreid-moz

### CODE REVIEW

File: etl/churn.kp/orig_src/Churn.ipynb [hunk 1]
```diff
@@ -473,126 +471,206 @@
...
+    "\n",
+    "env_date = environ.get('date', None)\n",
+    "if not env_date:\n",
+    "    raise ValueError(\"date not provided\")\n",
+    "\n",
+    "# If this job is scheduled, we need the input date to lag a total of \n",
+    "# 10 days of slack for incoming data + the 7 days used in churn computation, or 17 days\n",
+    "# from the current date. \n",
+    "week_start_date = snap_to_beginning_of_week(\n",
+    "    _datetime.strptime(env_date, \"%Y%m%d\") - timedelta(17),\n",
```
Comment (Dexterp37) [inline etl/churn.kp/orig_src/Churn.ipynb]: How does this behave when date is from an old week? Does it really compute the data for the day we're interested in? AFAIK Airflow could be passing old dates as well, not just the latest one.

Moreover, AFAICT the slack dance is already taken care of by |compute_churn_week|. Is there any other reason to do it here as well?

Comment (vitillo) [inline etl/churn.kp/orig_src/Churn.ipynb]: @Dexterp37 Please comment on the Markdown next time as it's more readable :)

Comment (Dexterp37) [inline etl/churn.kp/orig_src/Churn.ipynb]: Good point in making it clear in the preamble.

File: etl/churn.kp/knowledge.md [hunk 2]
```diff
@@ -428,12 +428,16 @@ def upload_to_s3(records, churn_outfile):
     transfer.upload_file(churn_outfile, churn_bucket, churn_s3,
                          extra_args={'ACL': 'bucket-owner-full-control'})
 
-    # Also upload it to the dashboard:
-    # Update the dashboard file
-    dash_bucket = "net-mozaws-prod-metrics-data"
-    dash_s3_name = "telemetry-churn/{}".format(churn_outfile)
-    transfer.upload_file(churn_outfile, dash_bucket, dash_s3_name,
-                         extra_args={'ACL': 'bucket-owner-full-control'})
+    # TODO: Re-enable uploading to the dashboard when cutover happends
```
Comment (acmiyaguchi) [inline etl/churn.kp/knowledge.md]: I will be disabling uploads to the dashboard while the csv location is still under `amiyaguchi/churn`. Once we verify that everything looks good, we can disable the ATMO script and enable this.

File: etl/churn.kp/orig_src/Churn.ipynb (hunk 1 above)
Comment (acmiyaguchi) [inline etl/churn.kp/orig_src/Churn.ipynb]: I've added a mention about the client latency and why the we aggregate data ~2 weeks in the past.

File: etl/churn.kp/knowledge.md [hunk 3]
```diff
@@ -433,182 +401,188 @@ def get_newest_per_client(df):
...
+        for r in records:
+            try:
+                fout.write(csv(r))
+                fout.write("\n")
+            except UnicodeEncodeError as e:
+                print("{}: Error writing line: {} // {}"
+                      .format(_datetime.utcnow(), e, r))
+        print "{}: finished writing lines".format(_datetime.utcnow())
+
+    # Now upload it to S3:
```
Comment (acmiyaguchi) [inline etl/churn.kp/knowledge.md]: Uploads to s3 can cause the job to fail if the object already exists. I should check for existence in both instances of uploading.

**Comment by Dexterp37** [comment]
Looks good to me.

**Pull request merged**